irodsadmin/management/commands/irodsorphans.py

```diff
@@ -25,184 +28,269 @@
-                collections.append(assay_path + '/' + RESULTS_COLL)
-                collections.append(assay_path + '/' + MISC_FILES_COLL)
-    return collections
-
-
-def get_study_collections(studies, irods_backend):
-    """Return a list of all study collection names."""
-    return [irods_backend.get_path(s) for s in studies]
-
-
-def get_zone_collections(irods_backend):
-    """
-    Return a list of all landing zone collection names that are not MOVED or
-    DELETED.
-    """
-    return [
-        irods_backend.get_path(lz)
-        for lz in LandingZone.objects.filter(
-            ~(Q(status=ZONE_STATUS_MOVED) & Q(status=ZONE_STATUS_DELETED))
+                )
+                continue
+
+            for assay in study.assays.all():
+                assay_table = study_tables['assays'][str(assay.sodar_uuid)]
+                assay_plugin = assay.get_plugin()
+                assay_path = self.irods_backend.get_path(assay)
+
+                if assay_plugin:
+                    for row in assay_table['table_data']:
+                        row_path = assay_plugin.get_row_path(
+                            row, assay_table, assay, assay_path
+                        )
+                        if row_path not in collections:
+                            collections.append(row_path)
+                    shortcuts = assay_plugin.get_shortcuts(assay)
+                    if shortcuts:
+                        for shortcut in shortcuts:
+                            collections.append(shortcut['path'])
+
+                    # Add default expected subcollections of assay collection
+                    collections.append(assay_path + '/' + TRACK_HUBS_COLL)
+                    collections.append(assay_path + '/' + RESULTS_COLL)
+                    collections.append(assay_path + '/' + MISC_FILES_COLL)
+        return collections
+
+    def _get_study_collections(self, studies):
+        """Return a list of all study collection names."""
+        return [self.irods_backend.get_path(s) for s in studies]
+
+    def _get_zone_collections(self):
+        """
+        Return a list of all landing zone collection names that are not MOVED or
+        DELETED.
+        """
+        return [
+            self.irods_backend.get_path(lz)
+            for lz in LandingZone.objects.filter(
+                ~(Q(status=ZONE_STATUS_MOVED) & Q(status=ZONE_STATUS_DELETED))
+            )
+        ]
+
+    def _get_project_collections(self):
+        """Return a list of all study collection names."""
+        return [
+            self.irods_backend.get_path(p)
+            for p in Project.objects.all().order_by('full_title')
+        ]
+
+    def _is_zone(self, collection):
+        """
+        Check if a given collection matches the format of path to a landing zone
+        collection.
+        """
+        projects_path = self.irods_backend.get_projects_path()
+        pattern = (
+            projects_path
```

Comment:
I think there might still be a problem here. We don't provide the start of line character `^` to the regex. This still gives a chance we have a false match if someone creates collections with the same name somewhere in the path. Please also check the other regexes for this.